client/post-editor/editor-sidebar/feedback-view.jsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes, PureComponent } from 'react';
+import { localize } from 'i18n-calypso';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import FeedbackSidebarHeader from './feedback-header';
+import FeedbackRequestForm from './feedback-request-form';
+import FeedbackList from './feedback-list';
+import SidebarFooter from 'layout/sidebar/footer';
+
+// TODO: Find a clearer word than "share" for the owner of feedback and use it in subcomponent and CSS classes
+class FeedbackView extends PureComponent {
+	static propTypes = {
+		translate: PropTypes.func,
+		close: PropTypes.func,
+		sharedLinks: PropTypes.array
+	}
+
+	constructor() {
+		super();
+		this.state = { openFeedbackCount: 0 };
+	}
+
+	onToggleFeedback = isOpen => {
+		const { openFeedbackCount } = this.state;
+
+		this.setState( {
+			openFeedbackCount: openFeedbackCount + ( isOpen ? 1 : -1 )
+		} );
+	};
+
+	haveShares() {
+		return this.props.sharedLinks.length > 0;
+	}
+
+	isAllFeedbackClosed() {
+		return this.state.openFeedbackCount === 0;
+	}
```

Comment:
often I like to try and pull all of the methods that I can out of the React component and make them functions in the module scope. granted, we probably won't have thousands of this component instantiated but it's an easy way to trim down the size of the component in memory and reduce GC churn. Also it makes these helpers easy to test.

```js
export const haveShares = ( { sharedLinks } ) => sharedLinks.length > 0;
export const isAllFeedbackClosed = ( { openFeedbackCount } ) => openFeedbackCount == 0;

// later

haveShares( this.props );
isAllFeedbackClosed( this.state );
```

of course, sometimes when we do this we recognize some patterns we could reuse…

```js
export const haveShares = links => links.length > 0;
export const isAllFeedbackClosed = count => count === 0;

// later

haveShares( this.props.sharedLinks );
isAllFeedbackClosed( this.state.openFeedbackCount );
```

then it can seem silly to have so much code extracted for such simple operations

```js
const { sharedLinks } = this.props;
const { openFeedbackCount } = this.state;

return (
	…
	{ openFeedbackCount === 0 && this.renderRequestForm() }
	{ sharedLinks.length && this.renderFeedbackList() }
	…
);
```

So in the end we may decide that the helper functions weren't all that much better at communicating what we're doing, though with `this.renderXYZ()` some of that meaning and coupling between the condition and the render is lost, whereas directly embedding them might complete the circle.

```js
	{ openFeedbackCount === 0 && (
		<div>
			<div className="…" />
			<FeedbackRequestForm />
		</div>
	) }
	{ sharedLinks.length && (
		<div…>
			…
		</div>
	) }
```